lib/rubocop/cop/metrics/line_length.rb

```diff
@@ -99,8 +99,8 @@ def ignore_cop_directives?
         end
 
         def allowed_uri_position?(line, uri_range)
-          uri_range.begin < max &&
-            (uri_range.end == line.length || uri_range.end == line.length - 1)
+          clean_line = line.gsub(/.[,"'}]$/, '')
```

Comment:
Generally regexps are a recipe for problems when processing code - e.g. this `,` might be something else. What exactly is triggering the problem?

Reply:
The regex was introduced to scrub the end of the string with only a set of terminal allowed characters. The problem is there are several characters at the end of the string that if not removed cause this logic to become very complex and not necessarily correct. 

For example, the trigger for this was a `",` at the end of a uri. Just the `,` or just the `"`  would be fine and covered by the `uri_range.end == line.length - 1`, but just adding `uri_range.end == line.length - 2` to the condition seems to avoiding the root problem.

I'd be for a better solution, but I'm not quite sure

Reply:
Yeah, but what you've done could break in something like a heredoc that ends with `",` on some line. Don't have much time to think about the proper solution right now, but I'm certain we can come up with some better (which would probably require a bigger change to the current logic).